sklearn_hierarchical/classifier.py

```diff
@@ -292,14 +291,44 @@ def _recursive_build_features(self, X, y, node_id, progress):
 
         return self.graph_.node[node_id]["X"]
 
+    def _extract_rows_csr(self, matrix, rows):
```

Comment:
This feels like a helper method that doesnt really belong in this class, maybe just a helper in `array.py`?